plugins/catalog-backend-module-azure-blobstorage/src/provider/AzureBlobStorageEntityProvider.ts

```diff
@@ -0,0 +1,192 @@
+        );
+      }
+      const taskRunner =
+        options.schedule ??
+        options.scheduler!.createScheduledTaskRunner(providerConfig.schedule!);
+
+      return new AzureBlobStorageEntityProvider(
+        providerConfig,
+        integration,
+        options.logger,
+        taskRunner,
+      );
+    });
+  }
+
+  private constructor(
+    private readonly config: AzureBlobStorageConfig,
+    integration: AzureBlobStorageIntegration,
+    logger: Logger,
+    taskRunner: TaskRunner,
+  ) {
+    this.integration = integration;
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+    this.storageClient = new BlobServiceClient(
+      `https://${integration?.config?.host}${integration?.config?.secretAccessKey}`,
+    );
+
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+  }
+
+  private createScheduleFn(taskRunner: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class: AzureBlobStorageEntityProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(`${this.getProviderName()} refresh failed`, error);
+          }
+        },
+      });
+    };
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  getProviderName(): string {
+    return `azureBlobStorage-provider:${this.config.accountName}`;
+  }
+
+  async refresh(logger: Logger) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    logger.info('Discovering Azure BlobStorage objects');
+
+    const keys = await this.listAllObjectKeys();
+    logger.info(`Discovered ${keys.length} Azure BlobStorage objects`);
+
+    const locations = keys.map(key => this.createLocationSpec(key));
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: locations.map(location => {
+        return {
+          locationKey: this.getProviderName(),
+          entity: locationSpecToLocationEntity({ location }),
+        };
+      }),
+    });
+
+    logger.info(
+      `Committed ${locations.length} Locations for Azure BlobStorage objects`,
+    );
+  }
+
+  private async listAllObjectKeys(): Promise<string[]> {
+    const keys: string[] = [];
+
+    const containerClient: ContainerClient =
+      this.storageClient.getContainerClient(this.config.containerName);
+    const blobs = containerClient.listBlobsFlat({ prefix: this.config.prefix });
+    for await (const blob of blobs) {
+      keys.push(`${this.config.containerName}/${blob.name}`);
```

Comment:
Feels like this will benefit from an optional filter of sorts. Either in the form of a callback function that, given a name, returns true or false; or in the form of a glob that you can use like the `minimatch` library to match on (see elsewhere in the repo for examples of that).

Not sure how common it will be that exactly all files are of interest., even irrespective of their extension